README.md

```diff
@@ -242,6 +242,26 @@ Linux wheels are built in the [`manylinux` docker images](https://github.com/pyp
 - The project directory is mounted in the running Docker instance as `/project`, the output directory for the wheels as `/output`. In general, this is handled transparently by `cibuildwheel`. For a more finegrained level of control however, the root of the host file system is mounted as `/host`, allowing for example to access shared files, caches, etc. on the host file system.  Note that this is not available on CircleCI due to their Docker policies.
 - Alternative dockers images can be specified with the `CIBW_MANYLINUX_X86_64_IMAGE` and `CIBW_MANYLINUX_I686_IMAGE` options to allow for a custom, preconfigured build environment for the Linux builds. See [below](#options) for more details.
 
+### Modern C++ standards
+
+Building Python wheels with modern C++ (C++11) requires a few tricks.
+
+Python 2.7 wheels cannot use C++17 at all, because Python 2.7 uses the `register` keyword, which is prohibited in C++17 standard.
```

Comment:
I think that all technical details should be moved to documentation and only linked from readme.

Reply:
Yes, definitely. But this comment is about the content, not about how to merge this into the new docs.

Reply:
`-Dregister=` or equivalent (that is, defining `register` as early as possible as a macro which expands to nothing) should also be an effective workaround for this unfortunate change in C++17.

Reply:
Good point! I think I was planning to do that, but then the compiler had a flag to just do so, so that was easier :-)

Bottom line: a lot of things are possible, but are not really supported in a straightforward way and need some hacks here and there. But I think we should be careful formulating what's not possible.

Reply:
I change code to test also c++17 against python 2.7. But not on windows.